### PR TITLE
ci/openshift-ci: Tweak test.sh to run with kubernetes

### PR DESCRIPTION
**Pull request by ldoktor**
Plain kubernetes do not support services and such, let's modify our test.sh (and related) to allow deploying and checking kata there, which might help testing things locally with kcli.

### CODE REVIEW

File: .ci/openshift-ci/run_smoke_test.sh [hunk 1]
```diff
@@ -40,13 +40,29 @@ hello_msg='Hello World'
 oc exec ${pod} -- sh -c "echo $hello_msg > $hello_file"
 
 info "Creating the service and route"
-oc apply -f ${script_dir}/smoke/service.yaml
-sleep 60
+if oc apply -f ${script_dir}/smoke/service.yaml; then
```
Comment (wainersm) [inline .ci/openshift-ci/run_smoke_test.sh:43]: The service API (https://kubernetes.io/docs/concepts/services-networking/service/) should work with kcli too. Maybe there is a firewall blocking the port 80 on the VM?

Comment (ldoktor) [reply .ci/openshift-ci/run_smoke_test.sh:43]: Services are supported, but routes are not. We can use ingress, but that one requires ingress controller, so using another pod seems simpler for the purpose of testing pod execution.

Comment (ldoktor) [reply .ci/openshift-ci/run_smoke_test.sh:43]: OK I found another way using service of type "NodePort", let me know which way you prefer.

**Review by wainersm — APPROVED** [review]
Cool! I liked the hostport approach better!

**Comment by wainersm** [comment]
@ldoktor missing the `Fixes`. At some point you will be automatically putting it! :)

**Comment by ldoktor** [comment]
Squashed and rebased, no code changes.

File: .ci/openshift-ci/run_smoke_test.sh [hunk 2]
```diff
@@ -21,7 +21,7 @@ oc apply -f ${script_dir}/smoke/${pod}.yaml || \
 # Check it eventually goes to 'running'
 #
 wait_time=600
-sleep_time=10
+sleep_time=5
```
Comment (gkurz) [inline .ci/openshift-ci/run_smoke_test.sh:24]: Optimizing the execution time of this script doesn't belong to this PR IMO.

File: .ci/openshift-ci/run_smoke_test.sh [hunk 3]
```diff
@@ -40,13 +40,25 @@ hello_msg='Hello World'
...
+    # Likely on OCP, use service
+    is_ocp=1
+    sleep 60
+    host=$(oc get route/http-server-route -o jsonpath={.spec.host})
+    port=80
+else
+    # Likely on plain kubernetes, test using another container
+    is_ocp=0
+    info "Failed to create service, likely not on OCP, trying via NodePort"
+    oc apply -f "${script_dir}/smoke/service_kubernetes.yaml"
```
Comment (gkurz) [inline .ci/openshift-ci/run_smoke_test.sh:53]: Any reason for not waiting 60 seconds like in the OCP case ?

File: .ci/openshift-ci/run_smoke_test.sh [hunk 4]
```diff
@@ -55,12 +67,11 @@ else
...
-info "Deleting resources created"
-oc delete -f ${script_dir}/smoke/service.yaml
-
-# Delete the pod.
+# Delete the resources.
 #
-info "Deleting the ${pod} pod"
+info "Deleting the resources"
+[ "$is_ocp" -eq 1 ] && oc delete -f ${script_dir}/smoke/service.yaml
+[ "$is_ocp" -eq 0 ] && oc delete -f ${script_dir}/smoke/service_kubernetes.yaml
```
Comment (gkurz) [inline .ci/openshift-ci/run_smoke_test.sh:74]: I'd personally express that with an `if then else`. Matter of taste :wink:

File: .ci/openshift-ci/run_smoke_test.sh [hunk 5]
```diff
@@ -55,12 +67,11 @@ else
 	info "HTTP server is unreachable"
 fi
 
-info "Deleting resources created"
-oc delete -f ${script_dir}/smoke/service.yaml
-
-# Delete the pod.
+# Delete the resources.
 #
-info "Deleting the ${pod} pod"
```
Comment (gkurz) [inline .ci/openshift-ci/run_smoke_test.sh:63]: I'd keep the `Deleting the ${pod} pod` message as it pairs with the `Creating the ${pod} pod` one.

**Review by gkurz — APPROVED** [review]
Thanks @ldoktor ! I made a few comments but nothing blocking.

File: .ci/openshift-ci/run_smoke_test.sh (hunk 2 above)
Comment (ldoktor) [reply .ci/openshift-ci/run_smoke_test.sh:24]: ack, it's a remnant from debugging. I'll put it into a different commit to have it revertable.

File: .ci/openshift-ci/run_smoke_test.sh (hunk 4 above)
Comment (ldoktor) [reply .ci/openshift-ci/run_smoke_test.sh:74]: ack, again, remnant from previsou version which handled things differently

File: .ci/openshift-ci/run_smoke_test.sh (hunk 3 above)
Comment (ldoktor) [reply .ci/openshift-ci/run_smoke_test.sh:53]: OCP uses service+route which might take a while to establish reliably while here we use nodeport directly. Anyway on top I added the `retry` so it would be possible to skip the 60s timeout on OCP as well as `curl` should handle that. I can add that as another commit if you want to.

File: .ci/openshift-ci/run_smoke_test.sh (hunk 5 above)
Comment (ldoktor) [reply .ci/openshift-ci/run_smoke_test.sh:63]: ack

**Comment by gkurz** [comment]
> Rebased on top of master, fixed the styling issues found by @gkurz and added 2 (optional) commits to include the unrelated changes. Let me know if I should remove them but they should slightly speedup the prep steps.

This PR is just fine ! Thanks @ldoktor !
